File: Projects/rosalind/HAMM.java
import java.io.*;
import java.util.*;
import java.util.stream.*;

public class HAMM {
	public static void main(String[] args) throws IOException {
		Scanner in = new Scanner(System.in);
		String seq = in.next(), pat = in.next();
		System.out.println(IntStream.range(0, seq.length()).filter(i -> seq.charAt(i) != pat.charAt(i)).count());
		in.close();
	}
}

File: Projects/rosalind/HAMM.kt
package rosalind

import java.io.IOException
import java.util.*
import java.util.stream.IntStream

object HAMM {
    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val stdin = Scanner(System.`in`)
        val seq = stdin.next()
        val pat = stdin.next()
        println(IntStream.range(0, seq.length).filter { i: Int -> seq[i] != pat[i] }.count())
        stdin.close()
    }
}